src/domain/users/users.repository.spec.ts

```diff
@@ -63,4 +72,115 @@ describe('UsersRepository', () => {
       expect(result).toEqual({ id: mockUser.id });
     });
   });
+
+  describe('deleteWalletFromUser', () => {
+    it('should remove a wallet from a user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockAuthenticatedWallet = walletBuilder().build();
+      mockWalletRepository.findOne.mockResolvedValueOnce(
+        mockAuthenticatedWallet,
+      );
+      mockWalletRepository.delete.mockResolvedValue({ affected: 1, raw: {} });
+
+      await usersRepository.deleteWalletFromUser({
+        authPayload,
+        walletAddress: addressToRemove,
+      });
+
+      expect(mockWalletRepository.delete).toHaveBeenCalledWith({
+        user: mockAuthenticatedWallet.user,
+        address: addressToRemove,
+      });
+    });
+
+    it('should throw an UnauthorizedException if the auth payload is empty', async () => {
+      const authPayload = new AuthPayload();
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(UnauthorizedException);
+    });
+
+    it('should throw a BadRequestException if there is only one wallet', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockAuthenticatedWallet = walletBuilder().build();
+      mockWalletRepository.findOne.mockResolvedValueOnce(
+        mockAuthenticatedWallet,
+      );
+      mockWalletRepository.count.mockResolvedValue(1);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: addressToRemove,
+        }),
+      ).rejects.toThrow(BadRequestException);
+    });
+
+    it('should throw an ConflictException if the user tries to remove the currently authenticated wallet', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      await expect(
+        usersRepository.deleteWalletFromUser({
+          authPayload,
+          walletAddress: authPayload.signer_address as `0x${string}`,
```

Comment:
Nit: casting can result in undesired side effects. I'd suggest throwing further up if `undefined` and removing this.

Reply:
Its strange because we are setting up the mock data ourselves and know that the signer address is there. What do you think about using `!` instead?

Reply:
Used a mock address instead